internal/checker/cloudformation/rules/general-security/verify_aws_lambda_function_dead_letter_queue.rego

```diff
@@ -0,0 +1,61 @@
+# METADATA
+# title: "Verify AWS Lambda function is configured for a Dead Letter Queue(DLQ)"
+# description: "Checks whether an AWS Lambda function is set up with a Dead Letter Queue(DLQ)."
+# scope: package
+# related_resources:
+# - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html
+# custom:
+#   id: CB_CFT_090
+#   severity: LOW
+package lib.cloudformation.CB_CFT_090
+
+import future.keywords.in
+
+is_valid {
+	supported_resources := ["AWS::Lambda::Function", "AWS::Serverless::Function"]
+	some resources in input.Resources
+	resources.Type in supported_resources
+}
+
+resource[type] {
+	some resource in input.Resources
+	is_valid
+	count(fail) > 0
+	type := resource.Type
+}
+
+resource[type] {
+	some resource in input.Resources
+	is_valid
+	count(pass) > 0
+	type := resource.Type
+}
+
+fail[block] {
+	is_valid
+	some resources in input.Resources
+	block := resources.Properties
+	not "TargetArn" in object.keys(block.DeadLetterConfig)
+}
+
```

Comment:
add another block for fail
fail[block] {
	is_valid
	some resources in input.Resources
	block := resources.Properties
	not "TargetArn" in object.keys(block.DeadLetterQueue)
}